service/auth.go

```diff
@@ -100,7 +100,16 @@ func (s *Service) unregisterClient(w http.ResponseWriter, r *http.Request) {
 	}
 	defer s.httpAudit("unregisterClient", data, w, r)
 
-	_, code, err := s.authHandler(w, r)
+	// If an admin client is not enabled, and self registration is not allowed,
+	// clients cannot unregister themselves.
+	if !s.cfg.API.Security.EnableAdmin && !s.cfg.API.Security.AllowSelfRegistration {
+		data.err = "enable_admin and allow_self_registration are both false"
```

Comment:
Not sure how much feedback you want to give in APIs -- I always err on the side of more info for API consumers (as long as it's not security-sensitive).

Reply:
This is not an user facing API as the client in this case is the plugin so I am not even sure how much of this would be surfaced but generally I think it would be best to log this information rather than returning it as this is not an actionable error from the client's perspective.

Reply:
Good point, I'll `log.Warn` the issue.